### PR TITLE
add hook for action policy

### PR DESCRIPTION
**Pull request by lHydra**
Added a hook for [Action Policy](https://github.com/palkan/action_policy)

The setup is similar to Pundit, so I decided not to split the README description into two sections

### CODE REVIEW

**Comment by lHydra** [comment]
@palkan посмотри момент с установкой пользователя, меня он немного смущает, что я через `@controller.class` это вызываю

File: lib/ckeditor/hooks/action_policy.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    # responds to each of the public methods here.
+    class ActionPolicyAuthorization
+      include Ckeditor::Helpers::Controllers
+
+      # See the +authorize_with+ config method for where the initialization happens.
+      def initialize(controller)
+        @controller = controller
+        # use ckeditor_current_user instead of default current_user so it works with
+        # whatever current user method is defined with Ckeditor
+        @controller.class.authorize :user, through: :ckeditor_current_user
```
Comment (palkan) [inline lib/ckeditor/hooks/action_policy.rb:18]: We can avoid this passing [context explicitly](https://actionpolicy.evilmartians.io/#/authorization_context?id=explicit-context):

```rb
authorize!(model_object,  context: {user: ckeditor_current_user})
```

File: lib/ckeditor/hooks/action_policy.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      end
+
+      # This method is called in every controller action and should raise an exception
+      # when the authorization fails. The first argument is the name of the controller
+      # action as a symbol (:create, :destroy, etc.). The second argument is the actual model
+      # instance if it is available.
+      def authorize(action, model_object = nil)
+        @controller.authorize!(model_object)
```
Comment (palkan) [inline lib/ckeditor/hooks/action_policy.rb:26]: Should there be an action?

```suggestion
        @controller.authorize!(model_object, to: :"#{action}?")
```

Comment (lHydra) [reply lib/ckeditor/hooks/action_policy.rb]: I don’t think it’s necessary. We can get the name of the action [dynamically](https://github.com/palkan/action_policy/blob/aa686b635e8705992084b43975a0738d41fd8479/lib/action_policy/rails/controller.rb#L46)

File: lib/ckeditor/hooks/action_policy.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+      # This method is called primarily from the view to determine whether the given user
+      # has access to perform the action on a given model. It should return true when authorized.
+      # This takes the same arguments as +authorize+. The difference is that this will
+      # return a boolean whereas +authorize+ will raise an exception when not authorized.
+      def authorized?(action, model_object = nil)
+        if action
+          @controller.allowed_to?(:"#{action}?",
+                                  model_object,
+                                  context: {user: @controller.current_user_for_action_policy})
```
Comment (palkan) [inline lib/ckeditor/hooks/action_policy.rb:35]: Why can't we use `ckeditor_current_user` directly? That would eliminate the need for ControllerExtension

```suggestion
                                  context: {user: @controller.ckeditor_current_user})
```

Comment (lHydra) [reply lib/ckeditor/hooks/action_policy.rb:35]: I decided to repeat the adapter interface for Pundit. I think this is done if we may need additional user manipulation for a specific adapter. But I agree that now it looks redundant. Do I need to change it [there](https://github.com/galetahub/ckeditor/blob/c6e4acc154c2cbbee5a6153b015451c7c4c5b595/lib/ckeditor/hooks/pundit.rb#L36) too?

**Comment by lHydra** [comment]
Hi @galetahub! Can you see the changed? Thanks!

**Pull request merged**